src/main/java/org/opensearch/ad/transport/EntityResultTransportAction.java

```diff
@@ -189,6 +190,25 @@ protected void doExecute(Task task, EntityResultRequest request, ActionListener<
             for (Entry<Entity, double[]> entityEntry : request.getEntities().entrySet()) {
                 Entity categoricalValues = entityEntry.getKey();
 
+                Map<String, String> attrValues = categoricalValues.getAttributes();
+                if (attrValues.containsKey(CommonName.EMPTY_FIELD)
+                    && detector.getCategoryField() != null
+                    && detector.getCategoryField().size() == 1) {
+                    // handle a request from a version before OpenSearch 1.1. Read EntityResultRequest(StreamInput in) for details.
+                    categoricalValues = Entity
+                        .createSingleAttributeEntity(
+                            detectorId,
+                            detector.getCategoryField().get(0),
+                            attrValues.get(CommonName.EMPTY_FIELD)
+                        );
+                }
+
+                attrValues = categoricalValues.getAttributes();
+                if (attrValues != null && attrValues.containsKey(CommonName.EMPTY_FIELD)) {
+                    // above remediation is not useful, skip the entity
+                    continue;
+                }
```

Comment:
Can you explain this if check? Based on lines 198-203 won't this always be true?

Reply:
lines 198-203 may change categoricalValues and thus the evaluation result might be different.

Reply:
Got it - but I'm still confused on what this check means and when it would be skipped. Can you help explain? I will also wait for your commit to update the check into a helper fn since that may help clarify for me too.

Reply:
will send new commits after testing.  The new code is like

```
/**
     * Whether the received entity comes from an node that doesn't support multi-category fields.
     * This can happen during rolling-upgrade or blue/green deployment.
     *
     * Specifically, when receiving an EntityResultRequest from an incompatible node,
     * EntityResultRequest(StreamInput in) gets an String that represents an entity.
     * But Entity class requires both an category field name and value. Since we
     * don't have access to detector config in EntityResultRequest(StreamInput in),
     * we put CommonName.EMPTY_FIELD as the placeholder.  In this method,
     * we use the same CommonName.EMPTY_FIELD to check if the deserialized entity
     * comes from an incompatible node.  If it is, we will add the field name back
     * as EntityResultTranportAction has access to the detector config object.
     *
     * @param categoricalValues deserialized Entity from inbound message.
     * @return Whether the received entity comes from an node that doesn't support multi-category fields.
     */
    private boolean isEntityeFromOldNodeMsg(Entity categoricalValues) {
        Map<String, String> attrValues = categoricalValues.getAttributes();
        return (attrValues != null && attrValues.containsKey(CommonName.EMPTY_FIELD));
    }
```

Let me know if the explanation is not clear.

Reply:
Cool, makes sense. Only thing I don't understand still is when the if check on line 207 will be true. Seems that it would get caught in the if check on line 194 and transformed first.

Reply:
you are right.  I cannot think of a case when the check on line 207 will be true.  will remove the branch around line 207.